variables.tf

```diff
@@ -904,3 +904,22 @@ variable "custom_container_apps" {
   }))
   default = {}
 }
+
+variable "container_app_identities" {
+  description = "Identities to assign to container app"
+  type = object({
+    type : optional(string)
+    identity_ids : optional(list(string))
+  })
+  default = {}
+
+  validation {
+    condition     = anytrue([var.container_app_identities.type == null && var.container_app_identities.identity_ids == null, contains(["SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned", ], coalesce(var.container_app_identities.type, " "))])
+    error_message = "Identity type must be one of the following: 'SystemAssigned', 'UserAssigned', or 'SystemAssigned, UserAssigned'!"
+  }
+
+  validation {
+    condition     = anytrue([var.container_app_identities.type == null && var.container_app_identities.identity_ids == null, length(coalesce(var.container_app_identities.identity_ids, [])) > 0])
```

Comment:
I don't think you need to `coalesce` `.identity_ids` because it's default is a `list[]`. You might be able to simplify and just do a length check of `>0` like you are doing

Reply:
Unfortunately this wouldn't work as by default it was null, not a `list[]`.